src/app/search/SEARCH.worker.ts

```
@@ -40,22 +55,245 @@ const {game, searchIndexHash} = await new Promise<WorkerMessageInputInit>(resolv
+
+    const res = await fetch(new URL(`/${toLowerCase(game)}/search-data.json?hash=${searchIndexHash}`, self.location.href), {
+        cache: 'force-cache',
+    });
     if (!res.ok) {
         try {
-            console.error('[SEARCH WORKER] Failed to fetch search index:', res.status, res.statusText, 'with result:', await res.text());
+            console.error('[SEARCH WORKER] Failed to fetch search index data:', res.status, res.statusText, 'with result:', await res.text());
         } finally {
             // eslint-disable-next-line no-unsafe-finally
-            throw new Error(`[SEARCH WORKER] Failed to fetch search index: ${res.status} ${res.statusText}`);
+            throw new Error(`[SEARCH WORKER] Failed to fetch search index data: ${res.status} ${res.statusText}`);
```

Biome (1.9.4)

[error] Unsafe usage of 'throw'.

'throw' in 'finally' overwrites the control flow statements inside 'try' and 'catch'.

(lint/correctness/noUnsafeFinally)